staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -974,28 +986,45 @@ func TestBookmarksOnResourceVersionUpdates(t *testing.T) {
 	expectedRV := 2000
 
 	wg := sync.WaitGroup{}
+	done := make(chan struct{})
+	errorCh := make(chan error, 1)
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
 		for {
-			event, ok := <-w.ResultChan()
-			if !ok {
-				t.Fatalf("Unexpected closed channel")
-			}
-			rv, err := cacher.versioner.ObjectResourceVersion(event.Object)
-			if err != nil {
-				t.Errorf("failed to parse resource version from %#v: %v", event.Object, err)
-			}
-			if event.Type == watch.Bookmark && rv == uint64(expectedRV) {
+			select {
+			case <-done:
```

Comment:
Sorry I do not follow why we need the done channel to exit the function? Aren't the existing `return` statements in this function enough?

Reply:
My bad 😄, this is actually not needed here (I had in mind this is multiple goroutines, but it's only one)